codex/node.nim

```diff
@@ -284,7 +284,13 @@ proc requestStorage*(
   let request = StorageRequest(
     ask: StorageAsk(
       slots: nodes + tolerance,
-      slotSize: (encoded.blockSize.int * encoded.steps).u256,
+      # TODO: Specify slot-specific size (as below) once dispersal is
+      # implemented. The current implementation downloads the entire dataset, so
+      # it is currently set to be the size of the entire dataset. This is
+      # because the slotSize is used to determine the amount of bytes to reserve
+      # in a Reservations
```

Comment:
The slotSize will be determined by the erasure coding mechanism (that has yet to be finalised), which determines the number of steps required. Because the erasure coding mechanism isn't finalised, we are simply downloading the entire dataset, and we need to ensure that the slotSize accurately represents the *max* size of the data that will need to be downloaded, because this is the amount of storage that will need to be carved off of the Availability.

Reply:
The network level erasure coding is there and usable, however, we're lacking the dispersal part which will distribute chunks to nodes in a way that keeps them uncorrelated, this part is yet to be finalized. For now, we can stub the dispersal mechanism and just simple split the dataset into even rows and each node gets a row.

Reply:
Fair point. I've created an issue for this: https://github.com/codex-storage/nim-codex/issues/557

Downloading the entire dataset seems to be very stable in all the testing (integration, distributed), so it would be good to have this handled in a separate PR so we can isolate the changes.